File: 백준/Silver/29715. 비밀번호 찾기/비밀번호 찾기.py
N, M = map(int, input().split())
X, Y = map(int, input().split())
x = 0
y = 0

for _ in range(M):
    a, b = map(int, input().split())
    if a != 0:
        x += 1
    else:
        y += 1

result = 1

if M == 0:
    for i in range(9, 9 - N, -1):
        result *= i
else:
    for i in range(y):
        result *= (N - x - i)
    for i in range(N - x - y):
        result *= (9 - x - y - i)
result = result * X + (int(result / 3) - (1 if result % 3 == 0 else 0)) * Y
print(result)

File: 백준/Silver/17219. 비밀번호 찾기/비밀번호 찾기.js
let fs = require('fs');
let input = fs.readFileSync(0, 'utf8').trim().split('\n');

const [N, M] = input[0].split(' ').map((v) => +v);
const info = input.slice(1, N + 1).map((i) => i.split(' '));
const site = input.slice(N + 1);

function solution(N, M, info, site) {
  const result = [];
  const map = new Map();

  info.forEach((i) => {
    const [s, p] = i;
    map.set(s, p);
  });

  site.forEach((s) => {
    result.push(map.get(s));
  });

  return result.join('\n').trim();
}

console.log(solution(N, M, info, site));
